src/ejabberd_sm_redis.erl

```diff
@@ -34,32 +34,24 @@ start(_Opts) ->
     RetVal.
 
 
--spec get_sessions() -> [ejabberd_sm:ses_tuple()].
+-spec get_sessions() -> [ejabberd_sm:session()].
 get_sessions() ->
     Keys = mongoose_redis:cmd(["KEYS", hash(<<"*">>)]),
     lists:flatmap(fun(K) ->
                           Sessions = mongoose_redis:cmd(["SMEMBERS", K]),
                           lists:map(fun(S) ->
-                                            Session = binary_to_term(S),
-                                            { Session#session.usr,
-                                              Session#session.sid,
-                                              Session#session.priority,
-                                              Session#session.info }
+                                            binary_to_term(S)
```

Comment:
I wonder how performant this is (mem, CPU) compared to tuples.

Reply:
`binary_to_term` returns a C array of size six with six pointers to six created-on-the-stop ERL_TERMs.
Then the next instruction allocates a new C array of size four, and copies four specific fields of those pointers from the previous array. It shouldn't need to copy the contents of the pointers, as they're immutable. So technically speaking all that is happening is an extra 4*64bit memory allocation, with the CPU it takes to allocate that and later on GC both arrays. Difference is probably so so minimal it really doesn't matter, but it's still totally unnecessary 🤷🏽‍♂️ Also, records are clearer than arbitrary tuples 😐

Reply:
Never mind, I thought for some reason that this PR changed the data stored in Redis, which is not the case.